lib/segment/src/index/hnsw_index/graph_links.rs

```diff
@@ -66,27 +74,41 @@ struct GraphLinksFileHeader {
     levels_count: u64,
     total_links_len: u64,
     total_offsets_len: u64,
-    offsets_padding: u64, // 0 or 4
+    offsets_padding: u64,
+    m: u16,
```

Comment:
Do we need to store `m` and `m0` here? Can we provide it from the graph where these values are already known? My concern is if links are uncompressed, `m` and `m0` are zeroes. Right now it's clear why, but this behavior may spend long hours of debugging in the future, I find it untrivial behavior

Reply:
As an alternative, if you wish to store `m` and `m0` anyway, you may create an additional structure with compression option. In this way you explicitly mark `m` and `m0` fields as compression options and say that in non-compression case developer cannot rely on `m`/`m0` fields